util/random/rand_test.go

```diff
@@ -84,3 +85,59 @@ func TestIncorrectSize(t *testing.T) {
 	dst := make([]byte, size+1)
 	cipher.XORKeyStream(dst, src)
 }
+
+func TestBits(t *testing.T) {
+	testCases := []struct {
+		bitlen uint // input bit length
+		exact  bool // whether the exact bit length should be enforced
+	}{
+		{bitlen: 128, exact: false},
+		{bitlen: 256, exact: true},
+		{bitlen: 512, exact: false},
+		{bitlen: 1024, exact: true},
+	}
+
+	for _, tc := range testCases {
```

Comment:
Nice pattern, but to improve test results reporting you'll want to use the Go sub-test pattern:
https://go.dev/blog/subtests

```
    for _, tc := range testCases {
        t.Run(fmt.Sprintf("bitlen: %d exact: %s", tc.bitlen, tc.exact), func(t *testing.T) {
            ...
        })
    }
```

same obviously applies to the other test :)